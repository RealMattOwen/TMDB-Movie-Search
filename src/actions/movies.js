import axios from 'axios';

export const requestResults = results => ({
    type: 'REQUEST_RESULTS',
    results
});

export const startRequestResults = userInput => {
    return dispatch => {
        return axios.get(`https://api.themoviedb.org/3/search/movie?query=${userInput}&api_key=375d33fe34920fdf8cb02b7ce3600ba7`)
            .then(res => {
                if (res.data.results.length) {
                    const movies = res.data.results.slice(0, 5);
                    dispatch(requestResults(movies));
                }
            })
            .catch(error => {
                console.log(error);
            });
    };
};

export const requestDetails = details => ({
    type: 'REQUEST_DETAILS',
    details
});

export const startRequestDetails = movieId => {
    return dispatch => {
        return axios.get(`https://api.themoviedb.org/3/movie/${movieId}?api_key=375d33fe34920fdf8cb02b7ce3600ba7`)
            .then(res => {
                const {
                    id,
                    backdrop_path: backgroundImage,
                    poster_path: coverImage,
                    title,
                    tagline: tagLine,
                    overview,
                    release_date: releaseDate,
                    runtime,
                    revenue: boxOffice,
                    vote_average: voteAverage
                } = res.data;
                const movie = {
                    id,
                    backgroundImage,
                    coverImage,
                    title,
                    tagLine,
                    overview,
                    releaseDate,
                    runtime,
                    boxOffice,
                    voteAverage
                };

                dispatch(requestDetails(movie));
            })
            .catch(error => {
                console.log(error);
            });
    };
};