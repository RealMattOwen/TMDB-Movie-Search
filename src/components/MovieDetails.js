import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const MovieDetails = props => (
    <div className="movie-container">
        <div className="content-container">
            <div className="movie-details">
                <div className="section">
                    <img src={`https://image.tmdb.org/t/p/original/${props.movies.movie.coverImage}?api_key=375d33fe34920fdf8cb02b7ce3600ba7`} />
                </div>
                <div className="section">
                    <h1 className="movie-title">{props.movies.movie.title}</h1>
                    <h2 className="movie-description">{props.movies.movie.tagLine}</h2>
                    <p>{props.movies.movie.overview}</p>
                    <div className="details">
                        <div className="detail">
                            <h3>Original Release:</h3>
                            <h2>{props.movies.movie.releaseDate}</h2>
                        </div>
                        <div className="detail">
                            <h3>Running Time:</h3>
                            <h2>{`${props.movies.movie.runtime} mins`}</h2>
                        </div>
                        <div className="detail">
                            <h3>Box Office:</h3>
                            <h2>{numeral(props.movies.movie.boxOffice).format('$0,0')}</h2>
                        </div>
                        <div className="detail">
                            <h3>Vote Average:</h3>
                            <h2>{`${props.movies.movie.voteAverage} / 10`}</h2>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
);

const mapStateToProps = (state, props) => ({
    movies: state.movies
});

export default connect(mapStateToProps, undefined)(MovieDetails);