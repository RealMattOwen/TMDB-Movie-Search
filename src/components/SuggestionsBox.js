import React from 'react';

const SuggestionsBox = props => (
    <ul className="suggestions-box">
        {props.userInput ? <li className="suggestion active">{props.userInput}</li> : <li className="suggestion active">''</li>}
        {props.movies ? props.movies.map(movie => {
            return <li
                className="suggestion"
                key={movie.id}
                id={movie.id}
                onClick={e => props.movieDetailsRequest(e.target.id)}
            >{movie.title}</li>
        }) : undefined}
    </ul>
);

export default SuggestionsBox;