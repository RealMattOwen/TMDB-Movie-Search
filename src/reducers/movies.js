const moviesReducerDefaultState = {
    results: [],
    movie: {
        id: 0,
        backgroundImage: '/c4zJK1mowcps3wvdrm31knxhur2.jpg',
        coverImage: '/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg',
        title: '',
        tagLine: '',
        overview: '',
        releaseDate: '',
        runtime: 0,
        boxOffice: 0,
        voteAverage: 0
    }
};

export default (state = moviesReducerDefaultState, action) => {
    switch (action.type) {
        case 'REQUEST_RESULTS':
            return {
                ...state,
                results: action.results
            };
        case 'REQUEST_DETAILS':
            return {
                ...state,
                movie: action.details
            };
        default:
            return state;
    }
};