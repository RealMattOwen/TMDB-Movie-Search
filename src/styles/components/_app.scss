#app {
	height: 100%;
}

.app {
	background-repeat: no-repeat;
	background-size: cover;
	height: 100%;
}