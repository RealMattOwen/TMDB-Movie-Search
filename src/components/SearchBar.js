import React from 'react';
import { connect } from 'react-redux';
import AutoComplete from 'material-ui/AutoComplete';
import { startRequestResults, startRequestDetails } from '../actions/movies';

export const SearchBar = props => (
    <AutoComplete
        dataSource={
            props.movies.results.map(result => {
                return {
                    title: result.title,
                    id: result.id
                };
            })
        }
        dataSourceConfig={{
            text: 'title',
            value: 'id'
        }}
        filter={(searchText, key) => true}
        fullWidth={true}
        hintStyle={{color: '#fff'}}
        hintText="Search Movie Title"
        inputStyle={{color: '#fff'}}
        onUpdateInput={props.startRequestResults}
        onNewRequest={(chosenRequest, index) => props.startRequestDetails(props.movies.results[index].id)}
        underlineFocusStyle={{borderBottomColor: 'rgba(1,207,115,1)'}}
    />
);

const mapStateToProps = (state, props) => ({
    movies: state.movies
});

const mapDispatchToProps = dispatch => ({
    startRequestResults: userInput => dispatch(startRequestResults(userInput)),
    startRequestDetails: movieId => dispatch(startRequestDetails(movieId))
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);