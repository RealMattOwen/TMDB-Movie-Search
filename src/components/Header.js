import React from 'react';
import SearchBar from './SearchBar';

const Header = () => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <a className="header__logo" href="https://www.themoviedb.org/">
                    <img src="https://www.themoviedb.org/assets/static_cache/27b65cb40d26f78354a4ac5abf87b2be/images/v4/logos/powered-by-rectangle-green.svg" />
                </a>
                <div className="search">
                    <SearchBar />
                </div>
            </div>
        </div>
    </header>
);

export default Header;