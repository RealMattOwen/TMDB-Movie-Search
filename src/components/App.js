import React from 'react';
import { connect } from 'react-redux';
import Header from './Header';
import MovieDetails from './MovieDetails';
import { startRequestDetails } from '../actions/movies';

export class App extends React.Component {
    componentDidMount() {
        this.props.startRequestDetails(11);
    }
    render() {
        return (
            <div className="app" style={{backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.45), rgba(0, 0, 0, 0.45)), url('https://image.tmdb.org/t/p/original${this.props.movies.movie.backgroundImage}')`}}>
                <Header />
                <MovieDetails />
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    movies: state.movies
});

const mapDispatchToProps = dispatch => ({
    startRequestDetails: userInput => dispatch(startRequestDetails(userInput))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);