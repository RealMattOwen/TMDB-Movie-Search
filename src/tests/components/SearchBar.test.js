import React from 'react';
import { shallow } from 'enzyme';
import movies from '../fixtures/movies';
import { startRequestResults, startRequestDetails } from '../../actions/movies';
import { SearchBar } from '../../components/SearchBar';

test('should render SearchBar correctly', () => {
    const wrapper = shallow(<SearchBar startRequestDetails={startRequestDetails} startRequestResults={startRequestResults} movies={movies} />);
    expect(wrapper).toMatchSnapshot();
});