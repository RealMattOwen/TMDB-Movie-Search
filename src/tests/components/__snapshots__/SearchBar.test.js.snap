// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar correctly 1`] = `
<AutoComplete
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  animated={true}
  dataSource={
    Array [
      Object {
        "id": 1,
        "title": "result 1",
      },
      Object {
        "id": 2,
        "title": "result 2",
      },
    ]
  }
  dataSourceConfig={
    Object {
      "text": "title",
      "value": "id",
    }
  }
  disableFocusRipple={true}
  filter={[Function]}
  fullWidth={true}
  hintStyle={
    Object {
      "color": "#fff",
    }
  }
  hintText="Search Movie Title"
  inputStyle={
    Object {
      "color": "#fff",
    }
  }
  menuCloseDelay={300}
  onNewRequest={[Function]}
  onUpdateInput={[Function]}
  open={false}
  openOnFocus={false}
  targetOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  underlineFocusStyle={
    Object {
      "borderBottomColor": "rgba(1,207,115,1)",
    }
  }
/>
`;
